File: problems/0150-evaluate-reverse-polish-notation/python/solution.py
def evalRPN(tokens):
    """Evaluate Reverse Polish Notation — https://leetcode.com/problems/evaluate-reverse-polish-notation/"""
    stack = []
    ops = {"+", "-", "*", "/"}
    for tok in tokens:
        if tok in ops:
            b = stack.pop()
            a = stack.pop()
            if tok == "+":
                stack.append(a + b)
            elif tok == "-":
                stack.append(a - b)
            elif tok == "*":
                stack.append(a * b)
            else:
                stack.append(int(a / b))  # truncate toward zero
        else:
            stack.append(int(tok))
    return stack[0]

File: problems/0150-evaluate-reverse-polish-notation/typescript/solution.ts
// Evaluate Reverse Polish Notation — https://leetcode.com/problems/evaluate-reverse-polish-notation/
export function evalRPN(tokens: string[]): number {
  const stack: number[] = [];
  const ops = new Set(["+", "-", "*", "/"]);
  for (const tok of tokens) {
    if (ops.has(tok)) {
      const b = stack.pop() ?? 0;
      const a = stack.pop() ?? 0;
      if (tok === "+") stack.push(a + b);
      else if (tok === "-") stack.push(a - b);
      else if (tok === "*") stack.push(a * b);
      else stack.push(Math.trunc(a / b));
    } else {
      stack.push(Number(tok));
    }
  }
  return stack[0];
}
